elixir/apps/web/lib/web/components/core_components.ex

```diff
@@ -870,41 +872,56 @@ defmodule Web.CoreComponents do
+        <thead class="text-xs text-gray-700 uppercase bg-gray-50 dark:bg-gray-700 dark:text-gray-400">
           <tr>
-            <th :for={col <- @col} class="p-0 pr-6 pb-4 font-normal"><%= col[:label] %></th>
-            <th class="relative p-0 pb-4"><span class="sr-only"><%= gettext("Actions") %></span></th>
+            <th :for={col <- @col} class="px-4 py-3">
+              <%= col[:label] %>
+              <.icon
+                :if={col[:sortable] == "true"}
+                name="hero-chevron-up-down-solid"
+                class="w-4 h-4 ml-1"
+              />
+            </th>
+            <th class="px-4 py-3">
+              <span class="sr-only"><%= gettext("Actions") %></span>
+            </th>
           </tr>
         </thead>
-        <tbody
-          id={@id}
-          phx-update={match?(%Phoenix.LiveView.LiveStream{}, @rows) && "stream"}
-          class="relative divide-y divide-zinc-100 border-t border-zinc-200 text-sm leading-6 text-zinc-700"
-        >
-          <tr :for={row <- @rows} id={@row_id && @row_id.(row)} class="group hover:bg-zinc-50">
+        <tbody id={@id} phx-update={match?(%Phoenix.LiveView.LiveStream{}, @rows) && "stream"}>
+          <tr :for={row <- @rows} id={@row_id && @row_id.(row)} class="border-b dark:border-gray-700">
             <td
-              :for={{col, i} <- Enum.with_index(@col)}
+              :for={{col, _i} <- Enum.with_index(@col)}
               phx-click={@row_click && @row_click.(row)}
-              class={["relative p-0", @row_click && "hover:cursor-pointer"]}
+              class={[
+                "px-4 py-3",
+                @row_click && "hover:cursor-pointer"
+              ]}
             >
-              <div class="block py-4 pr-6">
-                <span class="absolute -inset-y-px right-0 -left-4 group-hover:bg-zinc-50 sm:rounded-l-xl" />
-                <span class={["relative", i == 0 && "font-semibold text-zinc-900"]}>
-                  <%= render_slot(col, @row_item.(row)) %>
-                </span>
-              </div>
+              <%= render_slot(col, @row_item.(row)) %>
             </td>
-            <td :if={@action != []} class="relative w-14 p-0">
-              <div class="relative whitespace-nowrap py-4 text-right text-sm font-medium">
-                <span class="absolute -inset-y-px -right-4 left-0 group-hover:bg-zinc-50 sm:rounded-r-xl" />
-                <span
-                  :for={action <- @action}
-                  class="relative ml-4 font-semibold leading-6 text-zinc-900 hover:text-zinc-700"
+            <td :if={@action != []} class="px-4 py-3 flex items-center justify-end">
+              <button
+                id={"#{@row_id.(row)}-dropdown-button"}
+                data-dropdown-toggle={"#{@row_id.(row)}-dropdown"}
+                class="inline-flex items-center p-0.5 text-sm font-medium text-center text-gray-500 hover:text-gray-800 rounded-lg focus:outline-none dark:text-gray-400 dark:hover:text-gray-100"
```

Comment:
I feel like Tailwind modifiers are very noisy because of their prefixes and those lines that add horizontal scroll are hard to read. At the very least we can split them into groups to render them nicely:
```
class={[
  ["inline-flex", "items-center", "p-0.5", "font-medium", "rounded-lg"],
  ["text-sm", "text-center", "text-gray-500"], 
  ["hover:text-gray-800", "focus:outline-none"],
  ["dark:text-gray-400", "dark:hover:text-gray-100"]
]}
```
but this will also force us to come up with rules on how those should be grouped. Rules are bad unless there is something that enforces them while making your life easier so I guess we can come up with a helper similar to:
```
Interactive Elixir (1.15.2) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> ([
...(1)>   "inline-flex items-center p-0.5 font-medium rounded-lg",
...(1)>   {"text-", ["sm", "center", "gray-500"]},
...(1)>   {"hover:", "text-gray-800"},
...(1)>   {"focus:", "outline-none"},
...(1)>   {"dark:", "text-gray-400"},
...(1)>   {"dark:hover:", "text-gray-100"},
...(1)> ]
...(1)> |> Enum.flat_map(fn
...(1)>   {prefix, classes} when is_list(classes) ->
...(1)>     Enum.map(classes, fn class ->
...(1)>     "#{prefix}#{class}"
...(1)>     end)
...(1)>
...(1)>   {prefix, class} when is_binary(class) ->
...(1)>     ["#{prefix}#{class}"]
...(1)>
...(1)>   other -> [other]
...(1)> end))
["inline-flex items-center p-0.5 font-medium rounded-lg", "text-sm",
 "text-center", "text-gray-500", "hover:text-gray-800", "focus:outline-none",
 "dark:text-gray-400", "dark:hover:text-gray-100"]
```

I think it will prevent us from having horizontal scrolls and soft-force people to group classes into sections by Tailwind prefix. WDYT? 

cc @jamilbk

Reply:
What you've described makes sense to me.  I've never worked with Tailwind before, so I'm definitely open to anything.  @jamilbk any objections/thoughts?

I do hope that having components will help with some of that too.

Reply:
Hm, I find this a tad harder to read, debug, and edit than simply scrolling, or enabling line soft-wrap in the editor. It will also break searching and replacing for Tailwind classes.

I believe [HEEx automatically converts lists to space-delimited strings](https://hexdocs.pm/phoenix_live_view/0.18.6/Phoenix.Component.html#sigil_H/2-heex-extension-defining-attributes) so we may be able to get away with something simpler:

```elixir
<div class={~w[
  items-center
  rounded-lg
  ...
]}>
```

Reply:
@jamilbk With `~w` sigil it will work but you will need to think about grouping and then rules will be somewhere in the contribution guide. The main is that we should not use lines that are too long or raw lists (then the 20-item list will turn into 20 lines of code).

Reply:
Yeah agreed I think enforcing line length limits makes sense. I do think it's important to keep the class names whole so they're grep-able since it's such a common thing with Tailwind. Don't feel strongly about the solution here. Maybe we can use `~w` sigil and linebreak them manually until we have a better idea (or LiveView solves it for us :-P)

Reply:
Would this be acceptable?
```
<button
  id={"#{@row_id.(row)}-dropdown-button"}
  data-dropdown-toggle={"#{@row_id.(row)}-dropdown"}
  class={~w[
    inline-flex items-center p-0.5 text-sm font-medium text-center
    text-gray-500 hover:text-gray-800 rounded-lg focus:outline-none
    dark:text-gray-400 dark:hover:text-gray-100
  ]}
  type="button"
>
```

Reply:
I did a little reading on Tailwind last night as well and I agree, they do seem to heavily lean in to the idea that you should be using components and that's how you avoid having to look at every line of code when you're wanting to make any type of sweeping updates to your application.  So, I would agree that we should be striving to create and use as many components as we can.

Sprinkling in option 3 on your list also sounds like it could work well, but I don't have a strong opinion on that since I've not really worked with Tailwind before.

I was also reading this: https://tailwindcss.com/docs/customizing-colors#naming-your-colors
and was wondering if we wanted to use that instead of explicit color names everywhere.  Do either of you have strong thoughts/opinions on that? @jamilbk @AndrewDryga

Reply:
I like option (1) and (3). I feel like option (2) will evolve into the same spaghetti CSS that tailwind tries to prevent since it leads to precedence ambiguity and decouples where styles are defined from the element they're bound to, e.g.:

```
@layer components {
  h1, h2, h3, h4, h5 {
    @apply inline-flex;
    @apply items-center;
  }
  h1.hero {
    @apply font-lg;
  }
  h1 {
    @apply font-xl;
  }
}
```

I think we could reserve option (2) for places where we can't declare the classes in the template directly, like if we need to display safe HTML from another source. This is the only way to apply Tailwind classes to child elements AFAIK. I'm doing that in the website for [markdown content from GitHub here](https://github.com/firezone/firezone/pull/1740/files#diff-0cd40853b00c5dab22b8c3cf0e717a377279e406879b7ec84d8786301b27a23dR12).

@bmanifold We do name the colors we use in the website as `primary`, `accent`, and `neutral`. I didn't want to overload you with so much work renaming it all now, figured we could update all this in the polishing phase. But if you want to see what we're using in the website [look here for our brand colors](https://github.com/firezone/firezone/blob/master/website/tailwind.config.js#L70).

Reply:
@bmanifold I think using theme colors from Tailwind works. I just feel that we might end up having module attributes anyways because they will not just show the color, but also text weight and possibly alternative styling for dark mode, all in one place, so they can be reused together.

Let's agree to split everything into smaller components with `theme`/`style` attr then?